Assets/BossRoom/Scripts/Client/AnimationCallbacks/ClientAnimListener.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Client
+{
+    /// <summary>
+    /// Utility class used to listen for Animation-based events through Mecanim. This will invoke an Action when such
```

Comment:
```suggestion
    /// Utility class used to listen for Animation-based events (events coming from the animation itself and propagated to monobehaviour scripts) through Mecanim. This will invoke an Action when such
```

Reply:
Did I understand this right?

Reply:
Yes that's correct.

Reply:
> Trying to simplify instead of adding an additional layer of complexity. I just wanna make sure we've tried removing the cause of the issue, before adding patching.

I'll look into this simplification because you're right, it would remove the need for this one class.

Reply:
Class no longer needed! ClientCharacterVisualization handles anim events through Mecanim since it has been moved to the same GameObject as the Animator.

Reply:
So what was the move? And why is it there in the first place?

Reply:
So the child of the PlayerAvatar prefab, PlayerGraphics, used to only have Animator and NetworkAnimator components. The prefab that was instantiated for the graphics per character class used to contain ClientCharacterVisualization. We're able to move it to the same level as PlayerGraphics, but just had to update some of the references inside of the components of the instantiated prefab.